tests/common/dualtor/dual_tor_mock.py

```diff
@@ -483,3 +484,23 @@ def cleanup_mocked_configs(duthost, tbinfo):
     if is_t0_mocked_dualtor(tbinfo):
         logger.info("Load minigraph to reset the DUT %s", duthost.hostname)
         config_reload(duthost, config_source="minigraph", safe_reload=True)
+
+
+@pytest.fixture(scope="module")
+def restore_original_config_db(duthost, tbinfo):
```

Comment:
I believe we already have some fixtures to backup/restore config_db at `tests/common/fixtures/duthost_utils.py`, suggest  using/modifying these instead of adding new ones

Reply:
`cleanup_mocked_configs()` utilizes minigraph and does not necessarily restore the actual config state of the DUT before the test is run.
An example would be the prequisite of having `test_pretest.py` run (and make changes to `/etc/sonic/config_db.json`) before running many of the tests on some topos.

There are differences between the usual test config and minigraph generated config on t0. Replacing the original fixture may cause test issues as replacing it would mean the mocked dualtor tests will proceed in config used for normal t0 topology testing.

The original fixutre is also called much more frequently - once per setup function rather than once per testcase. I am unsure if each individual setup function requires a fresh minigraph generated config for test to work.

I can do some testing to see if replacing the original fixture with the new fixture will still pass the tests. Though even if there are no failures, I am not confident the dualtor mocked tests are run in the same state as with the original fixture.

Some confirmation from the reviewers will be great.

Reply:
why not simply reload the dut with the golden config? `config_reload` have an option to do that.

Reply:
In our test environment `/etc/sonic/golden_config_db.json` is empty and we run sonic-mgmt tests fine. I don't think it will good to assume all test environments will have `golden_config_db.json` available even if we populate it.

Reply:
Hi @justin-wong-ce, there is an option in the `config_reload` to use the running golden config to reload:
https://github.com/sonic-net/sonic-mgmt/blob/c98233132d57f4d517c9c1d9936ad5e29d0407c1/tests/common/config_reload.py#L184-L192

Reply:
Tested with `running_golden_config.json` and works well, thanks. PR updated.